### PR TITLE
Added glu_backward decomp

### PR DESCRIPTION
**Pull request by Chillee**
Requested in https://github.com/pytorch/pytorch/issues/93757

### CODE REVIEW

File: torch/_decomp/decompositions.py [hunk 1]
```diff
@@ -448,6 +448,23 @@ def _nll_loss_backward(
 
     return grad_input * grad_output
 
+
+@register_decomposition(aten.glu_backward)
+@pw_cast_for_opmath
+def glu_backward(grad_output: Tensor, self: Tensor, dim: int) -> Tensor:
```
Comment (mruberry) [inline torch/_decomp/decompositions.py:454]: I think we can just add this as a ref?

Comment (Chillee) [reply torch/_decomp/decompositions.py:454]: glu_backward doesn’t exist as a frontend op though, no? I thought we were gonna keep “aten-exclusive” ops as decomps.

Comment (Chillee) [reply torch/_decomp/decompositions.py:454]: Hmm... I'm not opposed in principle to lumping them together and tracking it that way, but the infra is currently not set up to handle "ATen-only ops". We don't have OpInfos for them, it's not really clear how they get registered in the `_refs` namespace, etc.

Considering that the only way to call these ops is through C++ today, it seems reasonable to keep them separate, although I'm open to other options.

cc: @ezyang

**Review by mruberry — APPROVED** [review]
Stamped

File: torch/_decomp/decompositions.py (hunk 1 above)
Comment (Chillee) [reply torch/_decomp/decompositions.py:454]: > we should stick them under _refs/internal or something

Oh yeah, I think we agreed that this is reasonable. But we should port everything in the `_decomps` folder to there simultaneously at some point imo.

Comment (ezyang) [reply torch/_decomp/decompositions.py:454]: Speaking of internal we should also have a tensor refs for Tensor method overrides

Comment (Chillee) [reply torch/_decomp/decompositions.py:454]: I did that in this PR :) https://github.com/pytorch/pytorch/pull/78697/files#diff-d65db46ea9f8125311f7e7d9acb23fd06cf4e681d35dea88afc4d396e8a87f7fR36

**Pull request closed without merging**